File: Java/백준/Silver/1072. 게임/게임.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        long x = sc.nextLong(); // 총 횟수
        long y = sc.nextLong(); // 이긴 횟수

        long z = (long) ((y * 100.0) / x);

        long s = 1L;
        long e = 1_000_000_000L;
        long answer = -1L;

        while (s <= e) {
            long mid = (s + e) / 2;
            long newZ = (long) (((y + mid) * 100.0) / (x + mid));

            if (newZ <= z) {
                s = mid + 1;
            } else {
                answer = mid;
                e = mid - 1;
            }
        }

        System.out.println(answer);
    }
}

File: Kotlin (JVM)/백준/Silver/1072. 게임/게임.kt
fun main() {
    var (x, y) = readln().split(" ").map { it.toLong() }
    val z = y * 100 / x

    var s = 1L
    var e = 1_000_000_000L
    var answer = -1L
    while (s <= e) {
        val mid = (s + e) / 2

        val newZ = (y + mid) * 100 / (x + mid)
        if (newZ == z) {
            s = mid + 1
        } else {
            answer = mid
            e = mid - 1
        }
    }
    println(answer)
}